src/window/macos.rs

```diff
@@ -0,0 +1,138 @@
+    }
+}
+
+lazy_static! {
+    // This height is in dpi-independent length, convert it to pixel length by multiply with scale factor.
+    static ref TITLEBAR_HEIGHT: f64 = MacosWindowFeature::calculate_titlebar_height();
+}
+
+pub struct MacosWindowFeature {
+    // Extra titlebar height in --frame transparecy. 0 in other cases.
+    extra_titlebar_height_in_pixel: u32,
+}
+
+impl MacosWindowFeature {
+    pub fn from_winit_window(window: &Window, mtm: MainThreadMarker) -> MacosWindowFeature {
+        let ns_window = if let RawWindowHandle::AppKit(handle) = window.raw_window_handle() {
+            unsafe { Id::retain(handle.ns_window as *mut NSWindow).unwrap() }
+        } else {
+            panic!("Not a appkit window.");
+        };
+
+        if let Ok(color) = &SETTINGS
+            .get::<WindowSettings>()
+            .background_color
+            .parse::<Color>()
+        {
+            error_msg!("neovide_background_color has now been deprecated. Use neovide_transparency instead if you want to get a transparent window titlebar. Please check https://neovide.dev/configuration.html#background-color-deprecated-currently-macos-only for more information.");
+
+            unsafe {
+                let [red, green, blue, alpha] = color.to_array();
+                let ns_background =
+                    NSColor::colorWithSRGBRed_green_blue_alpha(red, green, blue, alpha);
+                ns_window.setBackgroundColor(Some(&ns_background));
+            }
+        };
+
+        let mut extra_titlebar_height_in_pixel: u32 = 0;
+
+        let frame = SETTINGS.get::<CmdLineSettings>().frame;
+        if frame == Frame::Transparent {
+            unsafe {
+                let titlebar_click_handler = TitlebarClickHandler::new(mtm);
+
+                // Add the titlebar_click_handler into the view of window.
+                let content_view = ns_window.contentView().unwrap();
+                content_view.addSubview(&titlebar_click_handler);
+
+                // Set the initial size of titlebar_click_handler.
+                let content_view_size = content_view.frame().size;
+                titlebar_click_handler.setFrame(NSRect::new(
+                    NSPoint::new(0., content_view_size.height - *TITLEBAR_HEIGHT),
+                    NSSize::new(content_view_size.width, *TITLEBAR_HEIGHT),
+                ));
+
+                // Setup auto layout for titlebar_click_handler.
+                titlebar_click_handler.setAutoresizingMask(NSViewWidthSizable | NSViewMinYMargin);
+                titlebar_click_handler.setTranslatesAutoresizingMaskIntoConstraints(true);
+
+                extra_titlebar_height_in_pixel =
+                    Self::calculate_titlebar_height_in_pixel(window.scale_factor());
+            }
+        }
+
+        MacosWindowFeature {
+            extra_titlebar_height_in_pixel,
+        }
+    }
+
+    // Used to calculate the value of TITLEBAR_HEIGHT, aka, titlebar height in dpi-independent length.
```

Comment:
could be a doccomment (see [the reference](https://doc.rust-lang.org/stable/reference/comments.html#doc-comments) and/or [the book](https://doc.rust-lang.org/stable/book/ch14-02-publishing-to-crates-io.html#making-useful-documentation-comments)), so that it appears in the documentation generated by `cargo doc` as well:

```suggestion
    /// Used to calculate the value of TITLEBAR_HEIGHT, aka, titlebar height in dpi-independent length.
```